p2p/simulations/http.go

```diff
@@ -423,6 +464,66 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+//This function sets up filtering messages, if a non-empty filter string is provided
+//The filter string needs to be sent via the URL query string as 'filter=<strfilter>',
+//where strfilter has the format: [PROTOCOL]:[[MESSAGE-CODE],]
+//there needs to be at least one [PROTOCOL]:[MESSAGE-CODE] sequence; if multiple are present,
+//they need to be separated by a '-'
+//PROTOCOL must be a valid protocol for this to work (no validation here)
+//A message-code of '*' or '-1' are considered wildcards
+func (s *Server) setupFilter(strfilter string) error {
+	s.filters = make(map[MsgFilter]struct{})
+	s.wildcards = make(map[string]struct{})
```

Comment:
The filters and wildcards are per-request rather than per-server (so multiple stream requests would overwrite the filters of earlier requests). We need to make these per-request (perhaps just return them from this method).